symbolic-debuginfo/src/base.rs

```diff
@@ -450,13 +450,19 @@ pub struct FileInfo<'data> {
     name: Cow<'data, [u8]>,
     /// Path to the file.
     dir: Cow<'data, [u8]>,
+    /// Source code
+    source: Option<Cow<'data, [u8]>>,
 }
 
 impl<'data> FileInfo<'data> {
     /// Creates a `FileInfo` with a given directory and the file name.
     #[cfg(feature = "dwarf")]
-    pub fn new(dir: Cow<'data, [u8]>, name: Cow<'data, [u8]>) -> Self {
-        FileInfo { name, dir }
+    pub fn new(
```

Comment:
Instead of changing this constructor (and requiring a major version bump), how about adding a second `with_source` constructor?